display-points-using-clustering-feature-reduction/src/main/java/com/esri/arcgismaps/sample/displaypointsusingclusteringfeaturereduction/screens/MainScreen.kt

```diff
@@ -0,0 +1,117 @@
+import androidx.compose.material3.Switch
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.dp
+import com.esri.arcgismaps.sample.displaypointsusingclusteringfeaturereduction.components.ClusterInfoContent
+import com.esri.arcgismaps.sample.displaypointsusingclusteringfeaturereduction.components.ComposeMapView
+import com.esri.arcgismaps.sample.displaypointsusingclusteringfeaturereduction.components.MapViewModel
+import com.esri.arcgismaps.sample.sampleslib.components.BottomSheet
+import com.esri.arcgismaps.sample.sampleslib.components.LoadingDialog
+import com.esri.arcgismaps.sample.sampleslib.components.MessageDialog
+import com.esri.arcgismaps.sample.sampleslib.components.SampleTopAppBar
+import com.esri.arcgismaps.sample.sampleslib.theme.SampleTypography
+
+/**
+ * Main screen layout for the sample app
+ */
+@Composable
+fun MainScreen(sampleName: String, application: Application) {
+
+    // coroutineScope that will be cancelled when this call leaves the composition
+    val sampleCoroutineScope = rememberCoroutineScope()
+    // create a ViewModel to handle MapView interactions
+    val mapViewModel = remember { MapViewModel(application, sampleCoroutineScope) }
+
+    Scaffold(
+        topBar = { SampleTopAppBar(title = sampleName) },
+        content = {
+            Column(
+                modifier = Modifier
+                    .fillMaxSize()
+                    .padding(it),
+                horizontalAlignment = Alignment.CenterHorizontally,
+                verticalArrangement = Arrangement.Center
+            ) {
+                // composable function that wraps the MapView
+                ComposeMapView(
+                    modifier = Modifier.fillMaxSize().weight(1f),
+                    mapViewModel = mapViewModel
+                )
+                // Button to enable/disable featureReduction property
+                Row(
+                    modifier = Modifier
+                        .padding(12.dp)
+                        .fillMaxWidth(),
+                    verticalAlignment = Alignment.CenterVertically,
+                    horizontalArrangement = Arrangement.SpaceBetween
+                ) {
+                    Text(
+                        text = "Feature clustering",
+                        style = SampleTypography.bodyMedium
+                    )
+                    Switch(
+                        checked = mapViewModel.isFeatureReductionEnabled.value,
+                        onCheckedChange = {
+                            mapViewModel.toggleFeatureReduction()
+                        })
+                }
+
+                // display a MessageDialog if the sample encounters an error
+                mapViewModel.messageDialogVM.apply {
```

Comment:
I think this should be handled in the view model instead of in an UI element.

Reply:
I have followed this pattern to show MessageDialog from the MainScreen, from the other samples
CC: @shubham7109

Reply:
I will leave this to your judgement then